### DISCUSSION TITLE
langchain_llm vs other llms

### DISCUSSION THREAD

**Discussion by Hrishikesh007788** [sinaptik-ai/pandas-ai#552] [Q&A] [answered]
Documentation says that it also has support for langchain llms. How the experience will differ with
"langchain_llm = OpenAI(openai_api_key="my-openai-api-key")" compared to 
"llm = OpenAI(api_token='my-openai-api-key')"?

Does langchain_llm provide  more features?

**Comment by gventuri** [comment, marked as answer]
@Hrishikesh007788 there's not so much of a difference. Basically, if you have any specific reason to prefer the LangChain LLM, go for it, otherwise it's recommended to use the "native" OpenAI llm wrapper provided by PandasAI.

Basically LangChain LLMs have been implemented in order to allow users to use more LLMs.
